javaparser-core/src/main/java/com/github/javaparser/JavaParser.java

```diff
@@ -114,17 +118,20 @@ private GeneratedJavaParser getParserForProvider(Provider provider) {
         assertNotNull(start);
         assertNotNull(provider);
 
-        for (PreProcessor preProcessor : configuration.getPreProcessors()) {
-            provider = preProcessor.process(provider);
+        List<Processor> processors = configuration.getProcessors().stream().map(Supplier::get).collect(toList());
```

Comment:
Since some processors use preprocessing and postprocessing, and communicate between these two stages, we can't go instantiating all the preprocessor instantiating all the postprocessors. Some of them need to be the same instance between pre and postprocessing! This PR merges `PreProcessor` and `PostProcessor` into one `Processor` class and instantiates them all here, just before parsing.

Reply:
It is more awkward to use, since you can't use lambda syntax anymore.

Reply:
This is an API change. I don't feel to bad about it though, because I doubt many people have taken advantage of pre and postprocessors. Preprocessing is a highly specialized and rather awkward thing, and postprocessing can just be done in user code after parsing, so why use a postprocessor?